### PR TITLE
Use addNestedWhereQuery instead of "manually" merge

### PR DESCRIPTION
**Pull request by eriksulymosi**
**Changes**

use addNestedWhereQuery instead of "manually" merge

**Breaking changes**

no breaking changes

### CODE REVIEW

**Comment by spawnia** [comment]
Can you take a look at the PHPStan error? I think your code is correct, but you may have to do some extra work to convince PHPStan of it.

**Comment by eriksulymosi** [comment]
i'll check it

**Comment by eriksulymosi** [comment]
what do you think, might be a bug in larastan? :confused:

**Comment by spawnia** [comment]
If the definitions were more accurate, this could be avoidable. Until then - given you want this merged - we can work around the issue by extracting into variables and asserting the correct type through an annotation.

File: src/WhereConditions/WhereConditionsBaseDirective.php [hunk 1]
```diff
@@ -109,29 +109,26 @@ public function handleHasCondition(
...
-                        if ($condition) {
-                            $relatedModel = $this->nestedRelatedModel($model, $relation);
-
-                            $this->handleWhereConditions(
-                                $builder,
-                                $this->prefixConditionWithTableName(
-                                    $condition,
-                                    $relatedModel
-                                ),
+        /**@phpstan-ignore-next-line */
```
Comment (spawnia) [inline src/WhereConditions/WhereConditionsBaseDirective.php:112]:
```suggestion
        // @phpstan-ignore-next-line
```

**Comment by spawnia** [comment]
Thank you, nicely done

**Pull request merged**